dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/requests/McpRequest.java

```diff
@@ -37,19 +43,58 @@
  * @param method the method name
  * @param params the method parameters parsed as JSON
  */
-public record McpRequest(String jsonrpc,
-                         McpRequestId id,
-                         String method,
-                         JsonObject params) {
+public class McpRequest {
 
     private static final TraceComponent tc = Tr.register(McpRequest.class);
 
+    private String jsonrpc;
+
+    @JsonbTypeSerializer(McpRequestIdSerializer.class)
+    @JsonbTypeDeserializer(McpRequestIdDeserializer.class)
+    private RequestId id;
+
+    private String method;
+
+    @JsonbProperty("params")
+    private JsonObject params;
+
+    @JsonbCreator
+    public McpRequest(@JsonbProperty("jsonrpc") String jsonrpc,
+                      @JsonbProperty("id") RequestId id,
+                      @JsonbProperty("method") String method,
+                      @JsonbProperty("params") JsonObject params) {
+        this.jsonrpc = jsonrpc;
+        this.id = id;
+        this.method = method;
+        this.params = params;
+    }
+
+    @JsonbProperty("id")
+    @JsonbTypeSerializer(McpRequestIdSerializer.class)
+    @JsonbTypeDeserializer(McpRequestIdDeserializer.class)
+    public RequestId getId() {
+        return id;
+    }
+
+    public String getJsonrpc() {
+        return jsonrpc;
+    }
+
+    public String getMethod() {
+        return method;
+    }
+
+    public JsonObject getParams() {
+        return params;
+    }
+
     /**
      * Gets the request method as a {@link RequestMethod}.
      *
      * @return the matching {@link RequestMethod} enum value
      * @throws JSONRPCException if the method does not map to a {@code RequestMethod} value
      */
+    @JsonbTransient
```

Comment:
Are the changes in this class still needed now that you're adding your serializer classes to the Jsonb config?